pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -235,12 +236,25 @@ func GenerateTaskRun(
 		},
 	}
 
+	taskRunPodTemplate := &pod.PodTemplate{}
 	// Merge Build and BuildRun NodeSelectors, giving preference to BuildRun NodeSelector
 	taskRunNodeSelector := mergeMaps(build.Spec.NodeSelector, buildRun.Spec.NodeSelector)
 	if len(taskRunNodeSelector) > 0 {
-		expectedTaskRun.Spec.PodTemplate = &pod.PodTemplate{
-			NodeSelector: taskRunNodeSelector,
+		taskRunPodTemplate.NodeSelector = taskRunNodeSelector
+	}
+
+	// Merge Build and BuildRun Tolerations, giving preference to BuildRun Tolerations values
+	taskRunTolerations := mergeTolerations(build.Spec.Tolerations, buildRun.Spec.Tolerations)
+	if len(taskRunTolerations) > 0 {
+		// only support "NoSchedule" effect, make sure it is set
+		for i := range taskRunTolerations {
+			taskRunTolerations[i].Effect = corev1.TaintEffectNoSchedule
 		}
```

Comment:
Would not that be rather something for the validation logic ?

Reply:
Yes that makes more sense, changed